Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6734,7 +6734,7 @@ void WebPage::didChangeSelectionOrOverflowScrollPosition()
     // FIXME: We can't cancel composition when selection changes to NoSelection, but we probably should.
     if (frame->editor().hasComposition() && !frame->editor().ignoreSelectionChanges() && !frame->selection().isNone()) {
         frame->editor().cancelComposition();
-        discardedComposition();
+        discardedComposition(*frame->document());
```

Comment:
Ah, on closer inspection — it's not actually guaranteed that the frame's document is non null here after calling to `cancelComposition()` above. I'll add a null check here.